pkg/vere/main.c

```diff
@@ -256,14 +270,7 @@ _main_getopt(c3_i argc, c3_c** argv)
   {
     switch ( ch_i ) {
       case 5: {  //  urth-loom
-        c3_w lut_w;
-        c3_o res_o = _main_readw(optarg, u3a_bits_max+1, &lut_w);
-        if ( (c3n == res_o) || (lut_w < 20) ) {
-          fprintf(stderr, "error: --urth-loom must be >= 20 and <= %zu\r\n", u3a_bits_max);
-          return c3n;
-        }
-
-        u3_Host.ops_u.lut_y = lut_w;
+        _main_readw_loom(&u3_Host.ops_u.lut_y);
```

Comment:
The repetition of `--loom` argument parsing logic is irritating. But factoring it out like this changes the error message here (`--loom` vs `--urth-loom`). And it changes the control-flow on errors everywhere (previously returning `c3n` for invalid values, now continuing to parse). By the time you fix both, you're almost back to just duplicating the logic ... Any ideas?

Reply:
I noticed that they weren't identical, but hoped that I'd be told it didn't matter during review. How do you feel about have just one duplication: a special case for `urth-loom`?

Reply:
We could add a second parameter to `_main_readw_loom()` that is the argument name, so this call site would become `_main_readw_loom(&u3_Host.ops_u.lut_y, "--urth-loom")`, and the other call sites would become `_main_readw_loom(&u3_Host.ops_u.lom_y, "--loom")`.

Reply:
That works fine for the printf. I think it's important that the process exit with an error given invalid args, so you'll also need to propagate the return.

Reply:
I think it's better to just duplicate the logic in this case.  If we're worried about future changes needing to be applied in multiple places, we could `#define U3_MIN_LOOM 20`.

Reply:
Looks like @ashelkovnykov already `exit(1)`s on error in `_main_readw_loom()`, so adding a call to `u3_ve_usage()` before exiting should be sufficient to handle the error in case of invalid args.

Reply:
Quick recap - we have three cases:
1. `"--urth-loom"` and `return c3n`
2. `"--loom"` and `return c3n`
3. `"--loom"` and `exit(1)`

Proposed solution A:
- Change input of `_main_readw_loom()` to `const c3_c*` for custom `fprintf`
- Change return type of `_main_readw_loom()` to `c3_i`
- Return `0` on success
- Return `-1` on failure
- Each usage of `_main_readw_loom()` can determine what failure means

Proposed solution B:
- Leave `_main_readw_loom()` as is (bad input to `--loom` and `--urth-loom` will no longer show usage, just error message)
- Duplicate logic once for `--urth-loom` as special case

Proposed solution C:
- Make error message for `_main_readw_loom()` more generic:
  `fprintf(stderr, "error: loom arguments must be >= 20 and <= %u\r\n", u3a_bits_max);`

All of these are fine by me. I think Solution A is the most correct.

Reply:
When I first read this, I hadn't realized how many call sites there are to `_main_readw_loom()` -- more than enough to justify having a function.  I think we should go with solution A, since it makes the function useful from all of its call sites, without needing to carve out special cases or make error messages less distinct.

Reply:
Solution A is most correct, C is probably fine as well (so long as the process errors out on invalid args).

Reply:
Perfect - I think that's exactly what @mcevoypeter was describing in his two comments, so we've got agreement across the board. I'll edit the PR as soon as I'm able.

Reply:
Yep, Solution A sounds best to me too.